### PR TITLE
architecture: core: update mmu section

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Updates the MMU section to cover long-descriptor translation tables in a
subsection.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: architecture/core.rst [hunk 1]
```diff
@@ -428,7 +428,31 @@ MMU
...
 Translation tables
 ==================
-OP-TEE uses several L1 translation tables, one large spanning 4 GiB and two or
+
+OP-TEE supports two translation table formats:
+
+1. short-descriptor translation table format, availble on ARMv7-A and
+   ARMv8-A AArch32
+2. long-descriptor translation format, available on ARMv7-A with LPAE and
+   ARMv8-A
```
Comment (jforissier) [inline architecture/core.rst:437]: "Short descriptor format, availble...", "Long descriptor format, available..."
"LPAE (Large Physical Address Extension)" then drop the definition below.
s/availble/available/

File: architecture/core.rst [hunk 2]
```diff
@@ -428,7 +428,31 @@ MMU
...
+
+ARMv7-A without LPAE (Large Physical Address Extension) must use the
+short-descriptor translation table format only. ARMv8-A AArch64 must use
+the long-descriptor translation format only.
+
+Translation table format is static build time configuration option,
```
Comment (jforissier) [inline architecture/core.rst:443]: is a

File: architecture/core.rst [hunk 3]
```diff
@@ -428,7 +428,31 @@ MMU
...
+``CFG_WITH_LPAE``. The design around the translation table handling has
+been centered around these factors:
+
+1. Share translation tables between CPUs when possible to save memory
+   and simplify paging
+2. Support non-global CPU specific mappings to allow executing different
+   TAs in paralell.
```
Comment (jforissier) [inline architecture/core.rst:450]: parallel

File: architecture/core.rst [hunk 4]
```diff
@@ -483,6 +507,63 @@ context is activated.
...
+
+L1 and L2 translation tables are statically allocated and initialized at
+boot. Normally there is only one shared L2 table, but with ASLR enabled the
+virtual address space used for the shared mapping may need to use two
+tables. An unused entry in the L1 table is selected to point to the per
+thread L2 table. With ASLR configured this means that different per thread
+entry may be selected each time the system boots. Note that this entry will
+only point to a table when the per thread mapping is activated.
+
+The L2 translation tables in their turn points to L3 tables which uses the
```
Comment (jforissier) [inline architecture/core.rst:525]: point

**Review by jforissier — COMMENTED** [review]
With my comments addressed:
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by etienne-lms** [comment]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with Jérôme comments addressed, fwiw.

**Comment by jenswi-linaro** [comment]
Comments addressed and tags applied.

**Pull request merged**